conda_forge_tick/auto_tick.py

```diff
@@ -0,0 +1,930 @@
+            gdiff = eval_xonsh("git diff --name-only @(head_ref)...HEAD")
+
+            diffed_files = [
+                _
+                for _ in gdiff.split()
+                if not (
+                    _.startswith("recipe")
+                    or _.startswith("migrators")
+                    or _.startswith("README")
+                )
+            ]
+
+    pr_json: typing.Union[dict, bool]
+    if isinstance(migrator, MigrationYaml) and not diffed_files:
+        # spoof this so it looks like the package is done
+        pr_json = {"state": "closed", "merged_at": "never issued", "id": str(uuid4())}
+        ljpr = LazyJson(
+            os.path.join(migrator.ctx.parent.prjson_dir, str(pr_json["id"]) + ".json"),
+        )
+        ljpr.update(**pr_json)
+    else:
+        # push up
+        try:
+            pr_json = push_repo(
+                feedstock_dir,
+                migrator.pr_body(feedstock_ctx),
+                repo,
+                migrator.pr_title(feedstock_ctx),
+                migrator.pr_head(),
+                migrator.remote_branch(),
+            )
+
+        # This shouldn't happen too often any more since we won't double PR
+        except github3.GitHubError as e:
+            if e.msg != "Validation Failed":
+                raise
+            else:
+                print(f"Error during push {e}")
+                # If we just push to the existing PR then do nothing to the json
+                pr_json = False
+
+    # If we've gotten this far then the node is good
+    feedstock_ctx.attrs["bad"] = False
+    logger.info("Removing feedstock dir")
+    eval_xonsh(f"rm -rf {feedstock_dir}")
+    return migrate_return, pr_json
+
+
+def _requirement_names(reqlist):
+    """Parse requirement names from a list ignoring `None`
+    """
+    return [r.split()[0] for r in reqlist if r is not None]
+
+
+def _host_run_test_dependencies(meta_yaml):
+    """Parse the host/run/test dependencies of a recipe
+
+    This function parses top-level and `outputs` requirements sections.
+
+    The complicated logic here is mainly to support not including a
+    `host` section, and using `build` instead.
+    """
+    rq = set()
+    for block in [meta_yaml] + meta_yaml.get("outputs", []) or []:
+        req = block.get("requirements", {}) or {}
+        # output requirements given as list (e.g. openmotif)
+        if isinstance(req, list):
+            rq.update(_requirement_names(req))
+            continue
+
+        # if there is a host and it has things; use those
+        if req.get("host"):
+            rq.update(_requirement_names(req.get("host")))
+        # there is no host; look at build
+        elif req.get("host", "no host") not in [None, []]:
+            rq.update(_requirement_names(req.get("build", []) or []))
+        rq.update(_requirement_names(req.get("run", []) or []))
+
+    # add testing dependencies
+    for key in ("requirements", "requires"):
```

Comment:
This needs to be inside the blocks I think, since outputs could have their own tests.

Reply:
This is what used to be there